readme.markdown

```diff
@@ -3,44 +3,80 @@ CouchDB CI Setup
+* this [ASF Infra ticket](https://issues.apache.org/jira/browse/INFRA-10126).
 
-See the readme files in folder `baseboxes` (for docs on building the base boxes) and the section about Vagrant below (for docs on how to spin up the setup locally).
+*Remark: Throughout this repository we use the terms "master"/"worker" for the Jenkins build machines, whereas the Jenkins documentation uses the terms "master"/"slave".*
 
-Fair warning: This is very much work in progress.
+The main purpose of this repository is to provide a number of Docker containers that the ASF infrastructure team can use in their Jenkins setups and which are capable of building CouchDB. The idea is to provide containers for a number of different operating systems and Erlang versions to make sure CouchDB builds and runs on all supported setups.
 
-Current state:
+The current (rough) plan for the build matrix is this:
 
-- [x] install bare Jenkins master with Ansible
-- [x] install and configure nginx
-- [x] create CouchDB build job in Jenkins via Ansible
-- [x] switch to master-worker Jenkins setup
-- [x] use ntp server for master and workers
-- [ ] Use SCM sync plug-in to manage job configs
-    * http://stackoverflow.com/questions/27138043/jenkins-scm-sync-configuration-plugin-in-docker-wont-talk-to-github
-    * https://cburgmer.wordpress.com/2013/01/02/tracking-configuration-changes-in-jenkins/
-- [ ] enable auth for Jenkins
-- [ ] actually fetch CouchDB from VCS
-- [ ] all apt-get commands should pin a specific version, in the base box definition as well as in Ansible. How?
-- [ ] create an additional Ubuntu worker with an older Erlang version
-- [ ] create another base box (different linux distro) for a third worker
-- [ ] talk to Infra people
+**OS/Erlang**       | **R14B04** | **R16B03-1** | **17.5** | **18.x**
+--------------------|------------|--------------|----------|---------
+**Ubuntu 14.04**    | ?          | -            | -        | WIP
+**Ubuntu latest ?** | ?          | -            | -        | -
+**Debian 7**        | ?          | -            | -        | -
+**Debian 8**        | ?          | -            | -        | -
+**OS X latest**     | ?          | -            | -        | -
+**Free BSD**        | ?          | -            | -        | -
+**Windows**         | ?          | -            | -        | -
 
-*Remark: Throughout this repository we use the terms "master"/"worker" for the Jenkins build machines, whereas the Jenkins documentation uses the terms "master"/"slave".*
+### Open questions
+
+* AFAIK Erlang 14 support will be dropped soon-ish, so I'm not sure if it is worth the effort to do anything for that.
+* Which 18.x Erlang version is to be used? I heard someone saying 18.0 once, but that was before 18.1 and 18.2 were available, so I guess it makes more sense to always use the latest 18.x to see if changes in Erlang 18 breaks CouchDB.
```

Comment:
so perhaps there's two parts to this 1) confirming couchdb tests run with the standard dependencies from each of the various important OS/distro combinations and 2) confirming couchdb tests run against many versions of important dependencies like erlang itself, even if the OS in those cases remains the same.